CHANGELOG.md

```diff
@@ -6,6 +6,8 @@ FEATURES:
 IMPROVEMENTS:
 * Helm
   * Bump default Envoy version to 1.23.0. [[GH-1377](https://github.com/hashicorp/consul-k8s/pull/1377)]
+  * Added support for Consul API Gateway to read ReferenceGrant custom resources. This will require updating installed CRDs to include ReferenceGrant from the Gateway API v0.5 [Experimental Channel](https://gateway-api.sigs.k8s.io/concepts/versioning/#release-channels-eg-experimental-standard) if setting `apiGateway.enabled=true` [[GH-1299](https://github.com/hashicorp/consul-k8s/pull/1299)]
```

Comment:
There will be a few days between the release of this change and API Gateway v0.4. It would be nice if we could go ahead and say that ReferenceGrant is included in `github.com/hashicorp/consul-api-gateway/config/crd?ref=v0.4.0`, but that ref won't actually exist yet.